### ISSUE TITLE
SimpleConsumer: uncaught BufferUnderflowError

### ISSUE THREAD

**Issue by dpkp** [dpkp/kafka-python#229]
SimpleConsumer sends fetch requests for all partitions here:
https://github.com/mumrah/kafka-python/blob/master/kafka/consumer.py#L427-L431

But the "responses" generator tries to unpack the first message set immediately and will crash if the buffer size is too small:
https://github.com/mumrah/kafka-python/blob/master/kafka/protocol.py#L288

Because this is before we start the try/except buffer size logic, it just crashes on the first iteration.

I think the solution here is to expand the try/except block in `_fetch` to include the iteration step
```
for resp in responses:
```

**Comment by dpkp** [comment]
wontfix - SimpleConsumer and legacy protocol support both deprecated.
